src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -17,6 +17,7 @@ namespace D2L.CodeStyle.Analyzers.Immutability {
 	/// </summary>
 	internal sealed partial class ImmutabilityContext {
 		private readonly ImmutableDictionary<INamedTypeSymbol, ImmutableTypeInfo> m_extraImmutableTypes;
+		private readonly ImmutableHashSet<ITypeParameterSymbol> m_conditionalTypeParameters;

```

Comment:
Could just be `ITypeSymbol` and not parameter specific,, but :shrug: can be adjusted later if needed.